setup.py

```diff
@@ -11,7 +11,7 @@ def read(rel_path):
     here = os.path.abspath(os.path.dirname(__file__))
     # intentionally *not* adding an encoding option to open, See:
     #   https://github.com/pypa/virtualenv/issues/201#issuecomment-3145690
-    with open(os.path.join(here, rel_path), 'r') as fp:
+    with open(os.path.join(here, rel_path)) as fp:
```

Comment:
This is OK for now. But it will fail when non-ASCII characters are added in README.rst or `__init__.py`.
Please add `encoding="ascii"` or `encoding="utf-8"`.

Reply:
`encoding="utf-8"` is intentionally not addedd here for distutils compatibility. distutils does not write `METADATA` with UTF-8, but platform encoding, so using UTF-8 here would break packaging.

One possible denomitor is to use `encoding=None, error="replace"`.

Reply:
Good catch!
> 
> https://docs.python.org/3/library/functions.html#open
> The default encoding is platform dependent

We should probably go through all calls to `open()`  and `pathlib.Path.read_text()` add `encoding=something` everywhere. It may even be a good idea to add `encoding=locale.getpreferredencoding()` where the dependence on the OS and configuration is intentional. I think this is out of scope for this PR. As far as I understand `open(filename, 'r')` is exactly equivalent to `open(filename)` in python3

Reply:
If so, why not `encoding="ascii"`?

It prevents non-ASCII characters are added in README accidentally.

Reply:
Yeah, `encoding="ascii"` without `replace` would make sense here. I’d prefer this be made into a separate PR, however, and include reasoning in that PR to make it more discoverable for future maintainers.